File: web/src/types/node.ts
export interface NodeToken {
  token_id: string;
  token_type: string; // FBC/RPC/LOG/LIS
  port: number;
  protocol: string;
}

export interface Node {
  name: string;
  ip_address: string;
  status: 'offline' | 'online' | 'scanning' | 'error';
  tokens: NodeToken[];
}

File: go-backend/internal/models/node.go
package models

type NodeToken struct {
    TokenID   string `json:"token_id"`
    TokenType string `json:"token_type"` // FBC/RPC/LOG/LIS
    Port      int    `json:"port"`
    Protocol  string `json:"protocol"`
}

type Node struct {
    Name      string      `json:"name"`
    IPAddress string      `json:"ip_address"`
    Status    string      `json:"status"` // offline/online/scanning/error
    Tokens    []NodeToken `json:"tokens"`
}